.daily_canary

```diff
@@ -1,4 +1,4 @@
    ---     ___           ___
   (- -)   (= =)   |     Y   &  +-- 
```

Comment:
Seeing this here reminds me that I've encountered merge issues because of `.daily_canary`.  Is this a check to make sure that some CI has to run before this PR can be merged?

Reply:
The canary files (there are several) are used to kick off additional jobs, which aren't part of the main CI, but test useful things. The reason they aren't kicked off by default is because they take quite a long time to run.

Daily is run once a day, and includes things like a build with various sanitisers for example. To trigger that build on a specific PR, one can make a change (any change) to the .daily_canary file. The alternative would be to go to the ADO UI, and enter the branch ref in a form to kick off a build. It's possible but much more annoying than making a change to the file. The result is also not picked up by GitHub in that case.

The downside of this scheme, as you point out, is conflicts.

Reply:
If we moved to GitHub actions, we might be able to use more subtle triggers for this, perhaps labels on the PRs.

Reply:
In other words, there is no definition of what constitutes a correct merge of .daily_canary?!

Reply:
It's a button. It can be pressed (has received a change in this PR), or not pressed (has not received a change in this PR).

Any merge that gets the state you want on your PR (pressed or not pressed) is correct. Some pressed states are more artistic than others, but that's up to the person doing the pressing.

Reply:
We also have tooling to automerge the file (ie - splat it with your version): https://github.com/microsoft/CCF/pull/3675

Sadly GitHub doesn't support this, so when conflicts do occur this requires a client-merge.